Sources/History.swift

```diff
@@ -0,0 +1,369 @@
+    @objc public init(params: SearchParameters, highlightedText: String) {
+        self.params = params
+        self.highlightedText = highlightedText
+    }
+}
+
+/// A history that sits on the local device.
+///
+/// By default, the history sits purely in memory. If `filePath` is set, it is also persisted on disk.
+///
+@objc public class LocalHistory: NSObject, History {
+    // MARK: Properties
+    
+    /// The in-memory cache of the history's content.
+    private var queries: NSMutableOrderedSet = NSMutableOrderedSet()
+
+    /// The history's content, ordered by recency (read-only).
+    @objc public var contents: [String] {
+        return queries.array as! [String]
+    }
+    
+    /// Maximum number of entries allowed in this history. Default = 10.
+    @objc public var maxCount: Int = 10 {
+        didSet {
+            purge()
+        }
+    }
+    
+    /// Whether each query added automatically trigges a save. Default = `true`.
+    ///
+    /// + Note: The save will be performed asynchronously for better performance.
+    ///
+    @objc public var autosave: Bool = true
+    
+    /// File path to the history on disk.
+    /// If `nil` (default), the history will sit purely in memory.
+    @objc public var filePath: String?
+    
+    // MARK: Initialization
+    
+    /// Create a new, empty history that sits purely in memory.
+    public override init() {
+    }
+    
+    /// Create a new history that is persisted on disk.
+    ///
+    /// - parameter filePath: Path to the history file on disk.
+    ///
+    public init(filePath: String) {
+        self.filePath = filePath
+        super.init()
+        self.load()
+    }
+    
+    // MARK: History management
+    
+    /// Search the history.
+    ///
+    /// + Complexity: O(n): grows linearly with the size of the history.
+    ///   This should remain acceptable if the history is small (a few tens of entries).
+    ///
+    /// - parameter query: The searched query. By convention, an empty search matches the entire history
+    ///                    (up to `options.maxHits`).
+    /// - parameter options: Search options.
+    /// - returns: A list of matching hits in the history. They are ordered by recency (most recent first).
+    ///
+    @objc public func search(query: SearchParameters, options: HistorySearchOptions = HistorySearchOptions()) -> [HistoryHit] {
+        var hits: [HistoryHit] = []
+        let text = query.query ?? ""
+        let searchedText = normalize(text)
+        // Only search for the text at the beginning of a word.
+        let searchedPattern = "\\b" + searchedText
+
+        for query in queries {
+            // Stop when `maxHits` is reached.
+            if hits.count >= options.maxHits {
+                break
+            }
+            let queryString = query as! String
+            
+            var highlightedText: String
+            if searchedText.isEmpty { // empty search: everything matches, no highlighting necessary
```

Comment:
we can extract the logic of getHighlightedText in a separate function. 
and in order to take care of the "continue" statement, we can check if highlightedText is empty, then continue

Reply:
let's sync about it offline